crates/gitql-ast/src/function.rs

```diff
@@ -372,6 +380,54 @@ fn text_translate(inputs: Vec<Value>) -> Value {
     return Value::Text(text);
 }
 
+fn text_soundex(inputs: Vec<Value>) -> Value {
+    fn match_letter_to_int(letter: char) -> usize {
+        match letter {
+            'B' | 'F' | 'P' | 'V' => 1,
+            'C' | 'G' | 'J' | 'K' | 'Q' | 'S' | 'X' | 'Z' => 2,
+            'D' | 'T' => 3,
+            'L' => 4,
+            'M' | 'N' => 5,
+            'R' => 6,
+            _ => 0,
+        }
+    }
+
+    let text = inputs[0].as_text();
+    if text.is_empty() {
+        return Value::Text("".to_string());
+    }
+    
+    let letters_to_be_ignored = vec!['A', 'E', 'I', 'O', 'U', 'H', 'W', 'Y'];
+    //save the first letter
```

Comment:
No need for this comment because result it's for first letter only but used as result